### PR TITLE
Adding signature check and more tests to order-utils

### PR DESCRIPTION
**Pull request by dmosites**
- `orders.isValidOrder` now includes a call to `signatures.hasValidSignature`
- Adding tests to cover more variety of invalidity

### CODE REVIEW

File: utils/debugger/src/check-order.js [hunk 1]
```diff
@@ -20,8 +20,7 @@ const checkOrder = async (order, network) => {
 
   // Check the order has all necessary fields
   if (!orders.isValidOrder(order)) {
-    errors.push('Order structured incorrectly')
-    return errors
+    errors.push('Order structured incorrectly or signature invalid')
```
Comment (poiresel) [inline utils/debugger/src/check-order.js]: add push errors to line 27 as well for consistency.

File: utils/debugger/test/check-order.js [hunk 2]
```diff
@@ -274,9 +255,9 @@ describe('Orders', async () => {
 
     const errors = await checker.checkOrder(order, 'rinkeby')
 
-    assert.equal(errors.length, 2)
-    assert.equal(errors[0], 'sender no NFT approval')
-    assert.equal(errors[1], 'signer is not configured to receive NFTs')
+    assert.equal(errors.length, 3)
+    assert.equal(errors[1], 'sender no NFT approval')
```
Comment (poiresel) [inline utils/debugger/test/check-order.js]: assert the errors[0] as well

File: utils/order-utils/src/orders.js [hunk 3]
```diff
@@ -21,11 +21,14 @@ const signatures = require('./signatures')
 let nonce = 100
 
 const getLatestTimestamp = async () => {
-  return (await web3.eth.getBlock('latest')).timestamp
+  if (typeof web3 !== 'undefined') {
```
Comment (poiresel) [inline utils/order-utils/src/orders.js]: can you add a quick comment on why web3 may be undefined?

Comment (dmosites) [reply utils/order-utils/src/orders.js]: Not all environments have a global web3.

File: utils/debugger/src/check-order.js (hunk 1 above)
Comment (dmosites) [reply utils/debugger/src/check-order.js]: Do you mean to remove the return?

Comment (poiresel) [reply utils/debugger/src/check-order.js]: yes if the intended behavior is to proceed to the end and return all the errors.

**Pull request merged**